bin/tx-prover/src/proxy/mod.rs

```diff
@@ -14,31 +14,38 @@ use pingora_proxy::{ProxyHttp, Session};
 use tokio::sync::RwLock;
 use tracing::error;
 
-/// Timeout duration for the requests
-const TIMEOUT_SECS: Option<Duration> = Some(Duration::from_secs(100));
-
-/// Timeout duration for the connection
-const CONNECTION_TIMEOUT_SECS: Option<Duration> = Some(Duration::from_secs(10));
-
-/// Maximum number of items per queue
-const MAX_QUEUE_ITEMS: usize = 10;
-
-/// Maximum number of retries per request
-const MAX_RETRIES_PER_REQUEST: usize = 1;
+use crate::commands::ProxyConfig;
 
 /// Load balancer that uses a round robin strategy
-pub struct WorkerLoadBalancer(Arc<LoadBalancer<RoundRobin>>);
+pub struct LoadBalancer {
+    lb: Arc<PingoraLoadBalancer<RoundRobin>>,
+    timeout_secs: Duration,
+    connection_timeout_secs: Duration,
```

Comment:
What's the difference between `timeout_secs` and `connection-timeout_secs`?

Reply:
`timeout_secs` is used as an overall request timeout, measuring since the request starts until the response is sent. `connection-timeout_secs` is the timeout to establish a connection with a given worker.

Reply:
These seem to map to the proxy config's fields, but might be worth adding a short description for completeness I guess.